src/types/global/state.ts

```diff
@@ -2,7 +2,9 @@ import { StateNamesEnum } from '../../enums/state-names.enum';
 
 export interface IState {
   memberId: string,
-  state: StateNamesEnum,
-  step: number,
-  data: string
+  state: {
```

Comment:
Usually I tend to create interfaces for each object.
Meaning that the State object here could be instead an interface.
I do not have an idea of which could it be though since you are already in the state interface.

Reply:
I don't know either...